api/services/gameService.js

```diff
@@ -34,43 +34,33 @@ async function fetchSpectatorUsernames(gameId) {
 
 // Used to create fully populated game
 function tempUser(usr, points) {
-  this.pNum = usr.pNum;
-  this.hand = usr.hand;
-  this.points = points;
-  this.faceCards = usr.faceCards;
-  this.frozenId = usr.frozenId;
-  this.id = usr.id;
-  this.username = usr.username;
-
-  this.hand.sort(comapreByRankThenSuit);
-  this.points.sort(comapreByRankThenSuit);
-  this.faceCards.sort(comapreByRankThenSuit);
+  Object.entries(usr).forEach(([key,val]) => {
+    if(key === 'encryptedPassword'){
+      return;
+    }
+    if(key === 'points'){
+      this[key] = points;
+      return;
+    }
+    this[key] = val;
+  });
```

Comment:
You're totally right that we can take the same strategy here as with tempGame. I'd like to explore an approach that returns a new object rather than one that uses `this` as I think it's more clear. Then there are open questions about how to strip the encryptedPassword. Here's one way:

```suggestion
const res = {
  ...usr,
  points,
};
delete res.encryptedPassword;

```
Then we'd do the sorting below on `res` and return `res` at the end

Reply:
An alternative would be to use destructuring to pull the encryptedPassword off:

```
const {encryptedPassword, ...userDat} = usr;
const res = {
  ...userData,
  points
};

```

I think `delete` is probably clearer for being more explicit though

Reply:
Also let's rename the argument from `usr` to `user`. I thought I was being clever skipping a letter but it's annoying when you don't know when and whether it's usr vs user so just using user everywhere would be clearer